docker/util/image_util.sh

```diff
@@ -4,16 +4,17 @@ reset_cmd() {
     local original_image_name=$1
     local container_id=$2
     local output_image_name=$3
+    local docker_command=$4
```

Comment:
it seems to me you need to change the call to reset_cmd in https://github.com/bazelbuild/rules_docker/blob/master/docker/package_managers/run_install.sh.tpl#L25 to pass the expected docker_command?

Reply:
I did that.  Its here: https://github.com/bazelbuild/rules_docker/blob/ba0795d0fda5dcdd68107f37c0355145c10fd24e/docker/package_managers/run_install.sh.tpl#L26  
Am I missing something?

Reply:
RE Tests:
Can I get some more clarification on this?  It appears the tests you have are running from this script: https://github.com/bazelbuild/rules_docker/blob/master/testing/e2e.sh  And they are running targets that are actually in the `//tests/` directory.  You want me to put my target in the BUILD file next to my WORKSPACE in `//testing/custom_toolchain_flags`?  

Also, How do I run `file_test()` against bazel files like the one that ends up here: `bazel-out/darwin-fastbuild/bin/base_image.build`?

Thank you!

Reply:
> I did that. Its here:
Thanks for the clarification, not sure why I missed it.




> RE Tests:

So our test setup is quite complex. Lets try to only explain the bits that you actually have to move.
- You need to create a new directory under testing (similar to custom_toolchain_flags, but not the same one, call it custom_docker_flags)
- Directory has to have its own WORKSPACE and BUILD file to test your new features
- Tests will be run via GCB using a cloudbuild.yaml similar to this one: https://github.com/bazelbuild/rules_docker/blob/master/testing/download_pkgs_at_root/cloudbuild.yaml (and its basically as simple as the one pointed to here, just it lives under a different directory and has a different `dir` attr)
- You can test locally simply by running `bazel test ...` from this directory
- I will be doing the GCB setup (triggers), once the cloudbuild.yaml file is created in your PR

> Also, How do I run file_test() against bazel files like the one that ends up here: bazel-out/darwin-fastbuild/bin/base_image.build?

for this file the corresponding target should be `"//:base_image.build"`

Let me know if you have more questions. Thanks for your contributions!

Reply:
Excellent.  Thank you for the clarification and your patience.  Another quick question.  I see in `toolchain_configure()` that the `DOCKER_CONFIG` environment variable is used if set.  My question is: How is the rule getting the value of that env var?  According to the docs, the rule must specify that with the `environ` attr of the rule, but I don't see that happening.

Reply:
More clarification.. When you say:

> for this file the corresponding target should be "//:base_image.build"

This doesn't seem to be an output of `install_pkgs()`  it is generated but the final `ctx.actions.run()` only has `output_tar` as an output.  So running a test like: ```file_test(
    name = "test_custom_flag_install_pkgs_build",
    file = ":test_install_pkgs.build",
    regexp = "DOCKER_FLAGS=\"--log-level=info\"",
)``` returns errors like : ```//:test_custom_flag_install_pkgs_build: missing input file '//:test_install_pkgs.build'```

Reply:
> This doesn't seem to be an output of `install_pkgs()` it is generated but the final `ctx.actions.run()` only has `output_tar` as an output.

Yes, I just noticed this is not an explicit output so you cant really refer to it. You will need to make it an explicit output of `install_pkgs` by adding it here https://github.com/bazelbuild/rules_docker/blob/master/docker/package_managers/install_pkgs.bzl#L187 and changing the assignment of script [here](https://github.com/bazelbuild/rules_docker/blob/master/docker/package_managers/install_pkgs.bzl#L88) to point to your new output